agents/sorenson_dlm/dlm_agent.py

```diff
@@ -0,0 +1,259 @@
+        '''
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency')
+        if f_sample is None:
+            f_sample = self.f_sample
+
+        sleep_time = 1./f_sample - 0.01
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn("Could not start init because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'voltages',
+                    'data': {}
+                }
+                voltage_reading = self.dlm.read_voltage()
+                current_reading = self.dlm.read_current()
+                print('Voltage: {}'.format(voltage_reading))
+                print('Current: {}'.format(current_reading))
+                data['data']["voltage"] = np.float(voltage_reading)
+                data['data']["current"] = np.float(current_reading)
+
+                self.agent.publish_to_feed('voltages', data)
+                time.sleep(sleep_time)
+
+            self.agent.feeds['voltages'].flush_buffer()
+        return True, 'Acquistion exited cleanly'
+
+    def set_voltage(self, session, params=None):
+        """
+        Sets voltage of power supply:
+        Args:
+            voltage (float): Voltage to set.
+        
+        Examples:
+            Example of a client, setting the current to 1V:
+
+                client.set_voltage(voltage = 1.)
+
+        """
+
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            if acquired:
+                if self.over_volt == 0:
+                    return False, 'Over voltage protection not set'
+                elif float(params['voltage']) > float(self.over_volt):
+                    return False, 'Voltage greater then over voltage protection'
+                else:
+                    self.dlm.send_msg('SOUR:VOLT {}'.format(params['voltage']))
+            else:
+                return False, "Could not acquire lock"
+
+        return True, 'Set voltage to {}'.format(params['voltage'])
+
+    def set_over_volt(self, session, params=None):
+        """
+        Sets over voltage protection of power supply:
+        Args:
+            over_volt (int): Over voltage protection to set
+
+        Examples:
+            Example of a client, setting the overvoltage protection to 10V:
+
+                client.set_over_volt(over_volt = 10.)
+
+        """
+
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            if acquired:
+                print("ACQUIRED", params)
+                self.dlm.set_overv_prot(params['over_volt'])
```

Comment:
So there are several cases accounted for in `DLM.set_overv_prot` that result in the over voltage protection not actually being set, however they don't pass anything reflecting that in their returned values, and so the Agent can't determine that this is the case and will always return `True, 'Set over voltage protection to {}'.format(params['over_volt'])`. So it's always going to appear successful to the agent (and thus a client).

Reply:
Yes, returning False when `set_overv_prot` fails, and reacting accordingly in the agent sounds good to me.